### PR TITLE
Fix FileOutStreamIntegrationTest

### PR DESCRIPTION
**Pull request by voddle**
Removed an unsupported test.
Added new test for `FileOutStream.write(int)`
Fixed `AbstractFileOutIntegrationTest`

### CODE REVIEW

**Review by YichuanSun — COMMENTED** [review]
leave a comment, thanks.

File: dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java [hunk 1]
```diff
@@ -227,31 +240,4 @@ public void cancelWrite() throws Exception {
       Assert.assertEquals(0, worker.getUsedBytes());
     }
   }
-
-  @Test
-  public void getStatusBeforeClose() throws Exception {
```
Comment (YichuanSun) [inline dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java:232]: Can you explain the reason of this deletion?

Comment (voddle) [reply dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java:232]: have checked with mai, this test is no longer needed

File: dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java [hunk 2]
```diff
@@ -227,31 +235,4 @@ public void cancelWrite() throws Exception {
...
```
Comment (voddle) [inline dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java:232]: Due to the definition of related interface, this test is no longer needed.

File: dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java [hunk 3]
```diff
@@ -161,7 +169,7 @@ public void writeTwoByteArrays() throws Exception {
    */
   @Test
   public void longWrite() throws Exception {
-    AlluxioURI filePath = new AlluxioURI(PathUtils.uniqPath());
+    AlluxioURI filePath = new AlluxioURI(mPath.toString());
```
Comment (jiacheliu3) [inline dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java:172]: are you reusing the same file for every test? In `@After` I dont think the file is deleted between test cases so this kind of overwrite is really shady. Can you use a different path for every test case?

Comment (voddle) [reply dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java]: `mPath` is a child path of a `TemporaryFolder`, which as I understanded will be deleted when the test method end.

File: dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java [hunk 4]
```diff
@@ -161,7 +179,8 @@ public void writeTwoByteArrays() throws Exception {
    */
   @Test
   public void longWrite() throws Exception {
-    AlluxioURI filePath = new AlluxioURI(PathUtils.uniqPath());
+    String uniqPath = PathUtils.uniqPath();
```
Comment (jiacheliu3) [inline dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java:182]: `PathUtils.uniqPath()` generates a path with directories. Can you just use something easy like `new AlluxioURI(mRoot).join("longWrite");`?

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
left one comment PTAL thanks

File: dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java [hunk 5]
```diff
@@ -88,13 +81,30 @@ public void writeBytes() throws Exception {
...
+  @Test
+  public void writeInNonExistDirectory() throws Exception {
     CreateFilePOptions op = CreateFilePOptions.newBuilder().setWriteType(WritePType.CACHE_THROUGH)
         .setRecursive(true).build();
-    AlluxioURI filePath =
-        new AlluxioURI(PathUtils.concatPath(uniqPath, "file_" + MIN_LEN + "_" + mWriteType));
-    AlluxioURI parentPath = new AlluxioURI(uniqPath);
+    String uniqPath = PathUtils.uniqPath();
+    AlluxioURI filePath = new AlluxioURI(mRoot).join(uniqPath);
+    AlluxioURI parentPath = new AlluxioURI(mRoot.toString());
```
Comment (YichuanSun) [inline dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java:107]:
```suggestion
    AlluxioURI parentPath = new AlluxioURI(mRoot);
```

File: dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java [hunk 6]
```diff
@@ -184,7 +193,8 @@ public void longWrite() throws Exception {
    */
   @Test
   public void outOfOrderWrite() throws Exception {
-    AlluxioURI filePath = new AlluxioURI(PathUtils.uniqPath());
+    String uniqPath = PathUtils.uniqPath();
+    AlluxioURI filePath = new AlluxioURI(mRoot).join(uniqPath);
```
Comment (YichuanSun) [inline dora/tests/integration/src/test/java/alluxio/client/fs/io/FileOutStreamIntegrationTest.java:197]: BTW, as jiacheng said, if `PathUtils.uniqPath();` generates a directory, I don't think it is a good idea to use its return value as a file path.

**Review by YichuanSun — CHANGES_REQUESTED** [review]
Thanks! A few comments, PTAL.

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks for the work!

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**